mmhuman3d/models/body_models/utils.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+import numpy as np
+
+from mmhuman3d.utils.transforms import aa_to_rotmat, rotmat_to_aa
+
+
+def transform_to_camera_frame(global_orient, transl, pelvis, extrinsic):
+    """Transform body model parameters to camera frame.
+
+    Args:
+        global_orient (np.ndarray): shape (3, ). Only global_orient and
+            transl needs to be updated in the rigid transformation
+        transl (np.ndarray): shape (3, ).
+        pelvis (np.ndarray): shape (3, ). 3D joint location of pelvis
+            This is necessary to eliminate the offset from SMPL
+            canonical space origin to pelvis, because the global orient
+            is conducted around the pelvis, not the canonical space origin
+        extrinsic (np.ndarray): shape (4, 4). Transformation matrix
+            from world frame to camera frame
+    Returns:
+        (new_gloabl_orient, new_transl)
+            new_gloabl_orient: transformed global orient
+            new_transl: transformed transl
+    """
+
+    # take out the small offset from smpl origin to pelvis
+    transl_offset = pelvis - transl
+    T_p2w = np.eye(4)
+    T_p2w[:3, 3] = transl_offset
+
+    # camera extrinsic: transformation from world frame to camera frame
+    T_w2c = extrinsic
+
+    # smpl transformation: from vertex frame to world frame
+    T_v2p = np.eye(4)
+    global_orient_mat = aa_to_rotmat(global_orient)
+    T_v2p[:3, :3] = global_orient_mat
+    T_v2p[:3, 3] = transl
+
+    # compute combined transformation from vertex to world
+    T_v2w = T_p2w @ T_v2p
+
+    # compute transformation from vertex to camera
+    T_v2c = T_w2c @ T_v2w
+
+    # decompose vertex to camera transformation
+    # p_: new pelvis frame
+    # T_v2c = T_p_2c x T_v2p_
+    T_p_2c = T_p2w
+    T_v2p_ = np.linalg.inv(T_p_2c) @ T_v2c
```

Comment:
can we rename this to something like T_v2np? p_ is a bit ambiguous.

Reply:
Fixed in https://github.com/open-mmlab/mmhuman3d/pull/100/commits/13a42514f9eec927fce7412cc80d29dc4dbd88bd.